qa/L0_lint/pylintrc

```diff
@@ -22,6 +22,7 @@ disable=too-many-locals,
         attribute-defined-outside-init,
         global-statement,
         too-many-branches,
+        too-many-boolean-expressions,
```

Comment:
I presume this is to suppress a warning from https://github.com/NVIDIA/TransformerEngine/blob/a89e43c2c48023cfa539d62f6e351cc6b486ce60/transformer_engine/pytorch/attention.py#L562

I'd prefer keeping as many PyLint warnings as possible to keep our style disciplined. The offending if-statement seems sensible to me, so how about we locally suppress the warning instead with `# pylint: disable=too-many-boolean-expressions`.